sdk/python/test_picovoice.py

```diff
@@ -21,89 +18,101 @@
-                try:
-                    cpu_info = subprocess.check_output(['cat', '/proc/cpuinfo']).decode()
-                    cpu_part_list = [x for x in cpu_info.split('\n') if 'CPU part' in x]
-                    cpu_part = cpu_part_list[0].split(' ')[-1].lower()
-                except Exception as error:
-                    raise RuntimeError("Failed to identify the CPU with '%s'\nCPU info: %s" % (error, cpu_info))
-
-                if '0xb76' == cpu_part or '0xc07' == cpu_part or '0xd03' == cpu_part or '0xd08' == cpu_part:
-                    return os.path.join(
-                        os.path.dirname(__file__),
-                        '../../resources/rhino/resources/contexts/raspberry-pi/coffee_maker_raspberry-pi.rhn')
-                elif '0xd07' == cpu_part:
-                    return os.path.join(os.path.dirname(__file__),
-                                        '../../resources/rhino/resources/contexts/jetson/coffee_maker_jetson.rhn')
-                elif '0xc08' == cpu_part:
-                    return os.path.join(
-                        os.path.dirname(__file__),
-                        '../../resources/rhino/resources/contexts/beaglebone/coffee_maker_beaglebone.rhn')
-                else:
-                    raise NotImplementedError("Unsupported CPU: '%s'." % cpu_part)
-        elif platform.system() == 'Windows':
-            return os.path.join(os.path.dirname(__file__), '../../resources/rhino/resources/contexts/windows/coffee_maker_windows.rhn')
-        else:
-            raise ValueError("Unsupported system '%s'." % platform.system())
 
-    @classmethod
-    def _wake_word_callback(cls):
-        cls._is_wake_word_detected = True
+class PicovoiceMetaData:
+    def __init__(self, language, context, keywords):
+        self.language = language
+        self.keywords = keywords
+        self.context = context
+        self.reset()
 
-    @classmethod
-    def _inference_callback(cls, inference):
-        cls._inference = inference
+    def reset(self):
+        self.is_wake_word_detected = False
+        self.inference = None
+
+    def wake_word_callback(self):
+        self.is_wake_word_detected = True
+
+    def inference_callback(self, inference):
+        self.inference = inference
 
+
+class PicovoiceTestCase(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
-        cls._pv = Picovoice(
-            access_key=sys.argv[1],
-            keyword_path=pvporcupine.KEYWORD_PATHS['picovoice'],
-            wake_word_callback=cls._wake_word_callback,
-            context_path=cls._context_path(),
-            inference_callback=cls._inference_callback)
-
-        cls._is_wake_word_detected = False
-        cls._inference = None
+        metas = [
+            PicovoiceMetaData('en', 'coffee_maker', ['picovoice']),
+            PicovoiceMetaData('es', 'luz', ['manzana']),
+            PicovoiceMetaData('de', 'beleuchtung', ['heuschrecke'])]
+
+        cls._pvs = list()
```

Comment:
can you make this a `dict()` instead, with language as the key? That way instead of passing an index into `run_picovoice`, you are passing a language. This would make the tests more robust to change.

Reply:
I made some changes, please take another look